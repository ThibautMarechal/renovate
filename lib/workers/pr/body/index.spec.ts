import { platform } from '../../../../test/util';
import { RenovateConfig } from '../../../config';
import { getConfig } from '../../../config/defaults';

import { getPrBody } from '.';

describe('workers/pr/body', () => {
  let config: RenovateConfig;
  beforeEach(() => {
    jest.resetAllMocks();
    config = getConfig();
    platform.getPrBody = jest.fn((input) => input);
  });
  it('generate the message markdown', async () => {
    const branchName = 'test/markdown';
    const prBody = await getPrBody({
      ...config,
      branchName,
      upgrades: [
        {
          branchName,
          depType: 'dependencies',
          updateType: 'patch',
          depName: '@scope/package-with-link',
          sourceUrl: 'http://scope.com/package-with-link',
          displayFrom: '^10.1.1',
          displayTo: '^10.1.2',
        },
        {
          branchName,
          depType: 'dependencies',
          updateType: 'patch',
          depName: '@scope/package-without-link',
          displayFrom: '^10.1.1',
          displayTo: '^10.1.2',
        },
      ],
    });
    expect(prBody).toMatchSnapshot();
  });
});
