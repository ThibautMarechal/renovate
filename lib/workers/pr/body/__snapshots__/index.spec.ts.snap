// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workers/pr/body generate the message markdown 1`] = `
"This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [@&#8203;scope/package-with-link](http://scope.com/package-with-link) | dependencies | patch | \`^10.1.1\` -> \`^10.1.2\` |
| @&#8203;scope/package-without-link | dependencies | patch | \`^10.1.1\` -> \`^10.1.2\` |

---

### Renovate configuration

:date: **Schedule**: At any time (no schedule defined).

:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

:recycle: **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

:no_bell: **Ignore**: Close this PR and you won't be reminded about these updates again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Renovate Bot](https://github.com/renovatebot/renovate)."
`;
